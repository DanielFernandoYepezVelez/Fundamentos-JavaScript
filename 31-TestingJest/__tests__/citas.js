import Citas from '../js/class/Citas';

describe('Probar La Clase De Citas', () => {

    const citas = new Citas();
    const id = Date.now();

    test('Agregar Una Nueva Cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Daniel',
            telefono: '2867598',
            fecha: '10-12-2021',
            hora: '10:30',
            sintomas: 'Solo Duerme'
        };

        citas.agregarCita(citaObj);

        /* Prueba */
        expect(citas).toMatchSnapshot();
    });

    test('Actualizar Cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Nuevo Nombre',
            propietario: 'Daniel',
            telefono: '2867598',
            fecha: '10-12-2021',
            hora: '10:30',
            sintomas: 'Solo Duerme'
        };

        citas.editarCita(citaActualizada);
        expect(citas).toMatchSnapshot();
    });

    test('Eliminar Cita', () => {
        citas.eliminarCita(id);
        expect(citas).toMatchSnapshot();
    });
});