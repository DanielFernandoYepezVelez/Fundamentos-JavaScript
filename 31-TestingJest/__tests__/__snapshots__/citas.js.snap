// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar La Clase De Citas Actualizar Cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2021",
      "hora": "10:30",
      "id": 1619454554855,
      "mascota": "Nuevo Nombre",
      "propietario": "Daniel",
      "sintomas": "Solo Duerme",
      "telefono": "2867598",
    },
  ],
}
`;

exports[`Probar La Clase De Citas Agregar Una Nueva Cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2021",
      "hora": "10:30",
      "id": 1619454554855,
      "mascota": "Hook",
      "propietario": "Daniel",
      "sintomas": "Solo Duerme",
      "telefono": "2867598",
    },
  ],
}
`;

exports[`Probar La Clase De Citas Eliminar Cita 1`] = `
Citas {
  "citas": Array [],
}
`;
